### DISCUSSION TITLE
ESPHome Media Player not playing

### DISCUSSION THREAD

**Discussion by juptxn** [music-assistant/support#2527] [Q&A] [answered]
**The Problem:**
I flashed my ESP WT32-ETH01 (with a MAX98357A I2S Module) with ESPHome and added the media player in the yaml. It can play sound when I go to the ESP Device and choose a file to play via the media sources menu but when I try to play music through Music Assistant it sometimes gives me the error message **"Error: Resume queue requested but queue is empty"** or it just immediately stops the playback. 

Logs say **"2024-06-23 14:51:03.492 ERROR (MainThread) [music_assistant] Error doing task: Task exception was never retrieved
RuntimeError: anext(): asynchronous generator is already running"** right after trying to play music on the ESP Speaker.

I've also tried turning every option in the speaker settings on or off with no luck. Maybe somebody has an idea?

ESPHome YAML:
![Screenshot 2024-06-23 215744](https://github.com/music-assistant/hass-music-assistant/assets/124258985/9a52ef31-f492-4128-aea8-d28ceedbe419)

**What version of Music Assistant has the issue?**
2.0.7

**What version of the Home Assistant Integration have you got installed?**
2024.5.1

**Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?**
Yes

**What version of Home Assistant Core are your running**
2024.6.3

**What type of installation are you running?**
Home Assistant OS

**On what type of hardware are you running?**
x86 (64-bit)

**Comment by YahyaAlfayad** [comment]
I am having the same issue. I flashed an esp32 with esphome and itused to work on the previous versions of music assistant. After updating, I am getting:
```
2024-07-04 21:22:37.463 ERROR (MainThread) [music_assistant] Error doing task: Task exception was never retrieved
RuntimeError: anext(): asynchronous generator is already running
```

**Reply by OzGav** [reply, marked as answer]
To round this out for future readers. As of the time of this post this has been added to the docs

> ESPHome based Media Players are in general not recommended for playback of music. Short audio announcements and maybe even webradio could work but it is really not suitable for playing music from Music Assistant. TIP: you may have to enable the "fixed content length" HTTP profile in the player's settings. The ESPHome team is working on an improved media player which can be used on ESP32 modules with PSRAM, which will work correctly with Music Assistant.

**Reply by KE55ARD** [reply]
Would I be correct to assume this is still being worked on? I have a media player setup with PSRAM and it's not working for me currently.

**Reply by KE55ARD** [reply]
Ok I did actually get it to work! Bit crackly initially but playing well enough for now (ESP32-S3 with MAX98357 on custom board).
